jsapp/js/components/modalForms/dataAttachmentColumnsForm.es6

```diff
@@ -0,0 +1,226 @@
+      return {label: item.label, checked: true}
+    });
+    this.setState({columnsToDisplay: newList})
+  }
+  onBulkDeselect() {
+    let newList = this.state.columnsToDisplay.map((item) => {
+      return {label: item.label, checked: false}
+    });
+    this.setState({columnsToDisplay: newList})
+  }
+  onLoadAssetContentCompleted(response) {
+    if (
+      response.data_sharing?.fields.length > 0
+    ) {
+      this.setState({
+        isVirgin: false,
+        columnsToDisplay: this.props.generateColumnFilters(
+          this.props.fields,
+          response.data_sharing.fields,
+        ),
+      });
+    } else {
+      // empty `fields` implies all parent questions are exposed
+      this.setState({
+        isVirgin: false,
+        columnsToDisplay: this.props.generateColumnFilters(
+          this.props.fields,
+          response.content.survey,
+        ),
+      });
+    }
+  }
+  onPatchParentCompleted(response) {
+    this.setState({
+      isLoading: false,
+      columnsToDisplay: this.props.generateColumnFilters(
+        this.props.fields,
+        response.fields,
+      ),
+    });
+    this.props.onModalClose();
+  }
+  onPatchParentFailed() {
+    this.setState({isLoading: false});
+  }
+
+  onColumnSelected(newList) {
+    this.setState({columnsToDisplay: newList});
+  }
+
+  onSubmit(evt) {
+    evt.preventDefault();
+    this.setState({isLoading: true});
+    this.props.triggerParentLoading();
+
+    let fields = [];
+    var data = '';
+
+    this.state.columnsToDisplay.map((item) => {
+      if (item.checked) {
+        fields.push(item.label);
+      }
+    });
+
+    if (this.props.attachmentUrl) {
+      data = JSON.stringify({
+        fields: fields,
+        filename: this.props.filename,
+      });
+      actions.dataShare.patchParent(this.props.attachmentUrl, data);
+    } else {
+      data = JSON.stringify({
+        parent: this.props.parent.url,
+        fields: fields,
+        filename: this.props.filename,
+      });
+      actions.dataShare.attachToParent(this.props.asset.uid, data);
+    }
+  }
+
+  render() {
+    return (
+      <bem.FormModal__form m='data-attachment-columns'>
+        <div className='header'>
+          <span className='modal-description'>
+            {t('You are about to import ##PARENT_NAME##. Select or deselect in the list below to narrow down the number of questions to import.').replace('##PARENT_NAME##', this.props.parent.name)}
+          </span>
+
+          <div className='bulk-options'>
+            <span className='bulk-options__description'>
+              {t('Select below the questions you want to import')}
+            </span>
+
+            <div className='bulk-options__buttons'>
+              <a onClick={this.onBulkSelect}>
+                {t('select all')}
+              </a>
+
+              <span>
+                {t('|')}
+              </span>
+
+              <a onClick={this.onBulkDeselect}>
+                {t('deselect all')}
+              </a>
```

Comment:
Maybe something like this?

```javascript
{this.state.columnsToDisplay.length > 0 && (
  <a onClick={this.onBulkDeselect}>
    {t('deselect all')}
  </a>
)}
```

Reply:
On second thoughts, maybe this should rather behave more like the data table select-all and deselect-all check box?

<img width="204" alt="Screen Shot 2021-04-07 at 9 42 19 AM" src="https://user-images.githubusercontent.com/50016008/113903439-daaac000-9785-11eb-83e1-daf47a6387db.png">

Reply:
My implementation is from the mockups, but Leszek made the data table checkbox a component so I can integrate it in for sure. What do you think @Ig-Rebollo

Reply:
@duvld I don't know if you got an answer from Ignacio - if no, try asking him on Zulip :)

Reply:
There's a few changes Ignacio wanted to make about the look and feel in general, but we decided to do that after an initial release :D

Reply:
There's a few changes Ignacio wanted to make about the look and feel in general, but we decided to do that after an initial release :D